sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java

```diff
@@ -234,8 +234,17 @@ public BulkExecutor(CosmosAsyncContainer container,
                 return ((ItemBulkOperation<?>) itemOperation).getRetryPolicy().shouldRetry(operationResult).flatMap(
                     result -> {
                         if (result.shouldRetry) {
-                            groupSink.next(itemOperation);
-                            return Mono.empty();
+                            if (result.backOffTime == null || result.backOffTime.isZero()) {
```

Comment:
There are two places we are going to re-emit the operation.
I think this place probably only covers the 207 scenario, probably need to add similar logic for the batch 429 scenario, which is handled in retryOtherException

Reply:
Good catch - Thanks!

Reply:
Resolved